Zend/tests/accessors/unserialize.phpt

```diff
@@ -0,0 +1,46 @@
+--TEST--
+unserialize() with accessors
+--FILE--
+<?php
+
+class Test {
+    public $prop3 {
+        get { echo __METHOD__, "\n"; }
+        set { echo __METHOD__, "\n"; }
+    }
+
+    public function __construct(
+        public $prop1 { get; set; },
+        public $prop2 { get; },
+    ) {}
+}
+
+$test = new Test(1, 2);
+var_dump($s = serialize($test));
+var_dump($test2 = unserialize($s));
+$test2->prop3;
+$test2->prop3 = 42;
+
+// TODO: This should probably be forbidden.
```

Comment:
I'd agree as a maintainer of igbinary -- for computed properties such as `prop3` - `__serialize` and `__unserialize` could be added by applications in cases where data serialized for older versions of the classes was serialized with values for properties that later became magic accessors. (and saved to a cache/db to be read much later on)

- The alternative of supporting it by calling `set` sounds much worse, since computed `get` wouldn't be serialized and php is moving towards forbidding calling functions in unserialize on objects that haven't been fully initialized yet.

You've already forbidden overriding plain properties with accessor properties, so it doesn't seem likely for applications to set something they couldn't read with the same definition of a class and ancestors.

Reply:
> You've already forbidden overriding plain properties with accessor properties, so it doesn't seem likely for applications to set something they couldn't read with the same definition of a class and ancestors.

You are allowed to override a property with implicit accessors with one with explicit accessors though, so there might be a problem there...

Reply:
What I've implemented now is that this is not allowed for purely virtual properties, but is allowed if you had a parent with backing storage. In that case the child will also have backing storage, and I think that's reasonable in that the child could be making use of the parent (once there is some way to do that...)